kDrive/UI/Controller/Files/FileDetailViewController.swift

```diff
@@ -926,8 +926,8 @@ extension FileDetailViewController: ShareLinkTableViewCellDelegate {
 extension FileDetailViewController: RightsSelectionDelegate {
     func didUpdateRightValue(newValue value: String) {
         let right = ShareLinkPermission(rawValue: value)!
-        Task { [proxyFile = file.proxify()] in
-            _ = try await driveFileManager.createOrRemoveShareLink(for: proxyFile, right: right)
+        Task { [frozenFile = file.freezeIfNeeded()] in
```

Comment:
`Proxyfile` is not a DTO of a file, therefore is relatively empty and miss some important data.

Either I refactor to use _Frozen_ files, either I extend proxy files to be compliant DTOs of `File`.